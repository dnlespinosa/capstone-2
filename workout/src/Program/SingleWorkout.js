import React, { useEffect, useState } from 'react';

const SingleWorkout = ({ exercise }) => {

    return (
        <div>
            {exercise ? 
                <div className='SingleWorkout' style={{backgroundColor: 'white', border: '5px solid #2a2a2a', display: 'inline-block', color: '#2a2a2a', padding: '20px', margin: '10px 200px 10px 200px'}}>
                <div style={{display: 'inline-block'}}>
                    <h3>{exercise.exercise_name}</h3>
                    <p>Muscle Group: {exercise.target.Primary}</p>
                    <p>Instructions</p>
                    {exercise.steps.map(step => (
                        <li key={step}>{step}</li>
                    ))}
                </div>
                <video controls width="100%" style={{width: '50%', padding: '20px'}}>
                    <source src={exercise.videoURL[0]} type="video/mp4" />
                    Sorry, your browser doesn't support embedded videos.
                </video>
            </div> : 
            <p style={{color: 'white'}}>Loading Exercise...</p>}
        </div>
        
    )
}

export default SingleWorkout