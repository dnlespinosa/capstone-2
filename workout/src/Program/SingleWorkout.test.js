import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import SingleWorkout from "./SingleWorkout";
import { UserProvider } from "../testUtils";


// smoke test
it("renders without crashing", function () {
  render(
      <MemoryRouter>
        <UserProvider>
          <SingleWorkout />
        </UserProvider>
      </MemoryRouter>,
  );
});

// snapshot 
it('matches snapshot', function() {
    const { asFragment } = render(
            <MemoryRouter>
                <UserProvider>
                    <SingleWorkout />
                </UserProvider>
            </MemoryRouter>
        )
    expect(asFragment()).toMatchSnapshot();
  })




