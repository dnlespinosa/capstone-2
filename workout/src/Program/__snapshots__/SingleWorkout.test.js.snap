// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <p
      style="color: white;"
    >
      Loading Exercise...
    </p>
  </div>
</DocumentFragment>
`;
